### PR TITLE
Feature/document navigation components

### PR DESCRIPTION
**Pull request by Loque-**
- Add inline documentation to navigation components to help developers use them in their own application.
- Removed some example stories where the documented examples replaced them. 
- Added notes on HOC and nested CSS selectors

* [x] Documentation
* [x] Tests
* [x] Ready to be merged

### CODE REVIEW

File: components/related-items/README.md [hunk 1]
```diff
@@ -7,12 +7,36 @@ RelatedItems
...
+Simple
+```jsx
+import UnorderedList from '@govuk-react/unordered-list';
+import ListItem from '@govuk-react/list-item';
+import { asAnchor } from '@govuk-react/hoc';
+
+const AnchorTag = asAnchor('a');
+
+<RelatedItems>
+  <Header level={3}>Example header</Header>
```
Comment (taranchauhan) [inline components/related-items/README.md]: consider adding an import for Header as well.

```jsx
import Header from '@govuk-react/header';
```

File: components/related-items/src/index.js [hunk 2]
```diff
@@ -32,6 +30,35 @@ const StyledRelatedItems = styled('div')({
...
+ * Simple
+ * ```jsx
+ * import UnorderedList from '@govuk-react/unordered-list';
+ * import ListItem from '@govuk-react/list-item';
+ * import { asAnchor } from '@govuk-react/hoc';
+ *
+ * const AnchorTag = asAnchor('a');
+ *
+ * <RelatedItems>
+ *   <Header level={3}>Example header</Header>
```
Comment (taranchauhan) [inline components/related-items/src/index.js]: consider adding an import for Header as well.

```jsx
import Header from '@govuk-react/header';
```

File: components/related-items/README.md (hunk 1 above)
Comment (Loque-) [reply components/related-items/README.md]: Spotted, nice one

**Comment by Loque-** [comment]
Travis failed on chromatic, but passed everything else so merging into master! Hopefully we can get Chromatic back in to the mix as it was helping us a lot.

**Pull request merged**